data_steward/gcloud/bq/__init__.py

```diff
@@ -478,3 +484,44 @@ def table_exists(self, table_id: str, dataset_id: str = None) -> bool:
             return True
         except NotFound:
             return False
+
+    def serialize_jobs(self, job_list: list):
+        """
+        Waits on jobs until completion one by one
+
+        :param job_list: list of job_ids
+        """
+        for job_id in job_list:
+            job_info = self.get_job(job_id)
+            if job_info.state != 'DONE':
+                logging.info(f"Waiting on job {job_id} to complete")
+                job_info.result()
+
+    def wait_on_jobs(self,
+                     job_list: list = None,
+                     retry_limit: int = 300,
+                     backoff_limit: int = 2**8) -> list:
+        """
+        Waits on jobs until all are 'DONE' until backoff_limit is reached
+
+        :param job_list: list of job_ids. If not set, defaults to the last 10 jobs
+        :param retry_limit: Max time to wait in retry strategy
+        :param backoff_limit: Max time to wait in backoff strategy
+        :return jobs: list of incomplete job ids
+        """
+        incomplete_jobs = job_list
+        list_jobs_retry = retry.Retry(deadline=retry_limit)
+        backoff = 1
+        while incomplete_jobs and backoff <= backoff_limit:
+            bq_jobs = self.list_jobs(max_results=len(job_list) * 3,
+                                     state_filter='DONE',
```

Comment:
Have you tried this with the upper cased DONE?  The documentation indicates it should be lower cased.  https://googleapis.dev/python/bigquery/latest/generated/google.cloud.bigquery.client.Client.html?highlight=client#google.cloud.bigquery.client.Client.list_jobs

Reply:
Thanks for catching this, I'll try it out.

Reply:
This seems to work fine with the caps 'DONE' after testing.